### PR TITLE
Adds a new flag `commitmax` to the command `openchannel` of `lncli`

### PR DESCRIPTION
**Pull request by bjarnemagnussen**
### Description

This PR introduces a new flag `commitmax` for `openchannel` that for the channel capacity will use the whole available wallet balance up to the allowed maximum funding limit (currently ~0.16 btc) when opening a new channel. 

### Motivation

The command `sendcoins` has a flag `sweepall` that will spend all the coins from the wallet. However no such flag exists when opening new channels using `openchannel`.

Wanting to use the total available wallet balance in the local amount when opening a channel requires calculating the needed fee and subtracting it from the balance manually.

Further, it seems that this possibility has at least come up as most of the functionality to make this possible already exists, e.g.:
https://github.com/lightningnetwork/lnd/blob/80af2957ac37544f13b7945e5c8af1c834bce33b/lnwallet/chanfunding/assembler.go#L67-L72

### Edge Case

The `commitmax` flag makes use of the `CoinSelectSubtractFees` function to select funds. This implies that if the wallet balance is greater than the sum of the maximum allowed funding limit (~0.16 btc) and the fee, then for the new channel the capacity will be the maximum allowed (~0.16 btc) subtracted by the fee, instead of exactly the maximum allowed (~0.16 btc). This is a slightly wrong behaviour, as in this case it would have been possible to exactly max out the channel capacity and additionally have enough available funds to pay the fee.

I think a solution to this would require adding a field e.g. `CommitMax` to the `chanfunding.Request` and adding logic to the `ProvisionChannel` function inside `lnd/lnwallet/chanfunding/wallet_assembler.go` to distinguish between opening channels with `CommitMax` and without. But I think this is too much to specifically change for such an edge case.

### Disclaimer

I added this feature because I felt that such a flag was missing and could be useful. But I also implemented it just to get a better understanding of the `lnd` codebase. Let me know if this feature is useful or not for others.

### Testing

I am unsure if this change requires to add tests. But since this change does only use functionality that already existed, I don't see where tests should be added.

#### Pull Request Checklist

- [x] If this is your first time contributing, we recommend you read the [Code
  Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md)
- [x] All changes are Go version 1.12 compliant
- [x] The code being submitted is commented according to [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation)
- [ ] For new code: Code is accompanied by tests which exercise both
  the positive and negative (error paths) conditions (if applicable)
- [x] Code has been formatted with `go fmt`
- [x] For code and documentation: lines are wrapped at 80 characters
  (the tab character should be counted as 8 characters, not 4, as some IDEs do
  per default)
- [x] Running `make check` does not fail any tests
- [x] Running `go vet` does not report any issues
- [x] Running `make lint` does not report any **new** issues that did not
  already exist
- [x] All commits build properly and pass tests. Only in exceptional
  cases it can be justifiable to violate this condition. In that case, the
  reason should be stated in the commit message.
- [x] Commits have a logical structure according to [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure)

### CODE REVIEW

**Comment by Roasbeef** [comment]
A better way to do this is to use functionality added along side [`CreateSweepAllTx`](https://github.com/lightningnetwork/lnd/blob/master/sweep/walletsweep.go#L151) call to do it from a _coin selection_ perspective. In this case the "delivery" address is just the funding output. 

As implemented in this PR, it isn't concurrent safe, as the balance may shift when it's obtained on the CLI compared to when we actually go to fund the transaction. If the above function is used, then this isn't an issue as it locks all coin selection (and funding). 

Also a better name for this feature imo is something along the lines of `--fundall`. See this issue for some prior context as well: https://github.com/lightningnetwork/lnd/issues/619

**Comment by bjarnemagnussen** [comment]
Thanks for your input @roasbeef ! Those are some great insights you gave me. I will take another look at it and try to update this PR.

**Comment by bjarnemagnussen** [comment]
Will reopen PR as this has undergone major changes.

File: rpcserver.go [hunk 1]
```diff
@@ -1564,6 +1564,22 @@ func (r *rpcServer) OpenChannel(in *lnrpc.OpenChannelRequest,
...
+	// If the FundAll flag is set, ensure that the
+	// acceptable minimum local amount adheres to the
+	// amount to be pushed to the remote, and to current
+	// rules.
+	if in.FundAll {
+		// Use as minmimum local amount the larger of
+		// the amount to be pushed to the remote or
+		// the allowed minimum funding size.
+		if remoteInitialBalance >= minChanFundingSize {
+			localFundingAmt = remoteInitialBalance + 1
```
Comment (ziggie1984) [inline rpcserver.go:1577]: would be cool to add the information in the comment which was discussed below why the +1 is needed here

**Pull request closed without merging**